ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -512,11 +511,19 @@ private[ember] object Parser {
     }
   }
 
-  private def combineArrays[A: scala.reflect.ClassTag](a1: Array[A], a2: Array[A]): Array[A] = {
-    val buff = mutable.ArrayBuffer[A]()
-    buff.++=(a1)
-    buff.++=(a2)
-    buff.toArray
+  private[this] def appendByteArrays(a1: Array[Byte], a2: Chunk[Byte]): Array[Byte] = {
+    val res = new Array[Byte](a1.length + a2.size)
+    var ix = 0
+    while (ix < a1.length) {
+      res(ix) = a1(ix)
+      ix += 1
+    }
+    ix = 0
+    while (ix < a2.size) {
+      res(a1.length + ix) = a2(ix)
```

Comment:
@diesalbla actually we might want to check if `a2` is a `Chunk.ArraySlice` or `Chunk.ByteBuffer` and use specialized implementations for those. I think this `a2(ix)` call will be boxed.

Reply:
So, in the call from `recurseFind`, the `a2` chunk comes from the call to `read`, which would come from the implementations in FS2 for Socket. For some it seems to be using ByteBuffer,  so that would help.